Code/Framework/AzToolsFramework/AzToolsFramework/ComponentMode/ComponentModeSwitcher.h

```diff
@@ -0,0 +1,127 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#pragma once
+
+#include <AzToolsFramework/API/EntityCompositionNotificationBus.h>
+#include <AzToolsFramework/API/ViewportEditorModeTrackerNotificationBus.h>
+#include <AzToolsFramework/ComponentMode/EditorComponentModeBus.h>
+#include <AzToolsFramework/Entity/EntityTypes.h>
+#include <AzToolsFramework/ViewportUi/Button.h>
+#include <AzToolsFramework/ViewportUi/ViewportUiRequestBus.h>
+
+namespace AZ
+{
+    class Component;
+}
+namespace AzToolsFramework
+{
+    namespace ComponentModeFramework
+    {
+
+        enum class AddOrRemoveComponent : AZ::u8
+        {
+            Add,
+            Remove
+        };
+
+        // Struct containing relevant information about component for the switcher
+        struct ComponentData
+        {
+            ComponentData() = default;
+            ComponentData(AZ::EntityComponentIdPair pairId);
+
+            // Member variables
+            AZ::EntityComponentIdPair m_pairId; //!< Id of entity component pair.
+            AZ::Entity* m_entity = nullptr; //!< Pointer to entity associated with pairId.
+            AZ::Component* m_component = nullptr; //!< Pointer to component associated with pairId.
+            AZStd::string m_componentName; //!< Freindly name of component.
+            AZStd::string m_iconPath; //!< Path of component icon.
+            ViewportUi::ButtonId m_buttonId; //!< Button Id of switcher component.
+        };
+
+        struct Switcher
```

Comment:
update: This could use an API comment

Reply:
removed this struct as it isn't necessary